### ISSUE TITLE
No way to close a browser instance that was opened with connect()

### ISSUE THREAD

**Issue by ebidel** [puppeteer/puppeteer#989]
How does one close a browser instance that was opened by another script? Assuming you don't have access to that process.

Repro:
1. Run `node connect.js &`
2. Run `wsURL=<ws url printed to console in step 1> node connect.js`

```js
const puppeteer = require('puppeteer');

const browserWSEndpoint = process.env.wsURL || null;

(async () => {
  if (!browserWSEndpoint) {
    const browser = await puppeteer.launch();
    console.log(browser.wsEndpoint());
    return;
  }

  const browser2 = await puppeteer.connect({browserWSEndpoint});
  const page = await browser2.newPage();
  await page.goto('https://example.com');

  await browser2.close(); // Expected this to close the instance. I
})();
```

Thought about navigating to `about:quit` but that doesn't work: `Error: Failed to navigate: about:quit`

**Comment by aslushnikov** [comment]
On a first glance, this could be addressed with the `browser.close()` protocol method.

On a second thought though, the `browser.connect` method doesn't own the process, and thus shouldn't be able to close it. The service that started the browser should do the clients bookkeeping and manage browser instance lifetime.

What's your usecase?

**Comment by ebidel** [comment]
Protocol method sounds good. Making some examples for CDS, tried this, and it didn't work as expected.

 > The service that started the browser should do the clients bookkeeping and manage browser instance lifetime.

It's not always trivial to get back to the original service. One use case might be a long running chrome instance in cloud vm that you want to shut down after a period of time (cleanup / memory consumption issues). Your script or handler might be different than the one that launched Chrome.

Basically, if you can `connect()` to a browser, you should be able to close that browser. This makes `connect()` and `launch()` return the same thing. All methods continue to do the right thing.

**Comment by aslushnikov** [comment]
The `browser.close()` in its current form does make sense. Having both `browser.disconnect()` and `browser.close()` is confusing.

I like the `puppeteer.terminate(wsEndpoint)`. This would be doable with protocol method.

**Comment by ebidel** [comment]
Or what about allowing an opt param:`browser.close(wsEndpoint)`? That could even call `puppeteer.terminate` internally. My thinking is that it's simpler to have one API call to kill chrome.
